internal/injector/aspect/join/method.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package join
+
+import (
+	"github.com/datadog/orchestrion/internal/injector/node"
+	"github.com/dave/dst"
+	"github.com/dave/jennifer/jen"
+	"gopkg.in/yaml.v3"
+)
+
+type methodDefinition struct {
+	receiver TypeName
+	name     string
+}
+
+func MethodDefinition(receiver TypeName, name string) *methodDefinition {
+	return &methodDefinition{
+		receiver: receiver,
+		name:     name,
+	}
+}
+
+func (p *methodDefinition) Matches(chain *node.Chain) bool {
+	if chain.ImportPath() != p.receiver.path {
+		return false
+	}
+
+	node, ok := chain.Node.(*dst.FuncDecl)
+	if !ok || node.Recv == nil || node.Recv.NumFields() != 1 || node.Name.Name != p.name {
+		return false
+	}
+
+	if !p.receiver.Matches(node.Recv.List[0].Type) {
+		return false
+	}
+
+	return false
+}
```

Comment:
So this function always return false ?

Reply:
Since your tests are passing you are definitively missing at least one test

Reply:
Good catch! And now I wonder how the integration test suite manages to pass... o_O

Reply:
Okay the false positive was caused by a leftover `func init() {}` that was hiding the fact the instrumentation was not matching (had it matched, the test would have failed due to double-instrumentation)